calicoctl/commands/resources.go

```diff
@@ -58,104 +57,92 @@ func convertToSliceOfResources(loaded interface{}) []unversioned.Resource {
 		for i := 0; i < s.Len(); i++ {
-			r = append(r, convertToSliceOfResources(s.Index(i).Interface())...)
+			res = append(res, convertToSliceOfResources(s.Index(i).Interface())...)
 		}
 	case reflect.Struct:
-		// This is a resource or resource list.  If a resource, add to our return
-		// slice.  If a resource list, add each item to our return slice.
-		lr := loaded.(unversioned.Resource)
-		if strings.HasSuffix(lr.GetTypeMetadata().Kind, "List") {
+		// This is a resource or resource list. If a resource, add to our return
+		// slice. If a resource list, add each item to our return slice.
+		lr := loaded.(runtime.Object)
+		if strings.HasSuffix(lr.GetObjectKind().GroupVersionKind().Kind, "List") {
 			items := reflect.ValueOf(loaded).Elem().FieldByName("Items")
 			for i := 0; i < items.Len(); i++ {
-				r = append(r, items.Index(i).Interface().(unversioned.Resource))
+				res = append(res, items.Index(i).Interface().(runtime.Object))
 			}
 		} else {
-			r = append(r, lr)
+			res = append(res, lr)
 		}
 	case reflect.Ptr:
-		// This is a resource or resource list.  If a resource, add to our return
-		// slice.  If a resource list, add each item to our return slice.
-		lr := reflect.ValueOf(loaded).Elem().Interface().(unversioned.Resource)
-		if strings.HasSuffix(lr.GetTypeMetadata().Kind, "List") {
+		// This is a resource or resource list. If a resource, add to our return
+		// slice. If a resource list, add each item to our return slice.
+		lr := reflect.ValueOf(&loaded).Elem().Interface().(runtime.Object)
+		if strings.HasSuffix(lr.GetObjectKind().GroupVersionKind().Kind, "List") {
 			items := reflect.ValueOf(loaded).Elem().FieldByName("Items")
 			for i := 0; i < items.Len(); i++ {
-				r = append(r, items.Index(i).Interface().(unversioned.Resource))
+				res = append(res, items.Index(i).Addr().Interface().(runtime.Object))
 			}
 		} else {
-			r = append(r, lr)
+			res = append(res, lr)
 		}
 	default:
 		panic(errors.New(fmt.Sprintf("unhandled type %v converting to resource slice",
 			reflect.TypeOf(loaded).Kind())))
 	}
 
-	log.Infof("Returning slice: %v", r)
-	return r
+	log.Infof("Returning slice: %v", res)
+	return res
 }
 
 // getResourceFromArguments returns a resource instance from the command line arguments.
-func getResourceFromArguments(args map[string]interface{}) (unversioned.Resource, error) {
+func getResourceFromArguments(args map[string]interface{}) (runtime.Object, error) {
 	kind := args["<KIND>"].(string)
 	name := argutils.ArgStringOrBlank(args, "<NAME>")
-	node := argutils.ArgStringOrBlank(args, "--node")
-	workload := argutils.ArgStringOrBlank(args, "--workload")
-	orchestrator := argutils.ArgStringOrBlank(args, "--orchestrator")
-	resScope := argutils.ArgStringOrBlank(args, "--scope")
+	namespace := argutils.ArgStringOrBlank(args, "<NS>")
+
 	switch strings.ToLower(kind) {
 	case "node", "nodes", "no", "nos":
-		p := api.NewNode()
-		p.Metadata.Name = name
-		return *p, nil
+		n := api.NewNode()
```

Comment:
shouldn't need this switch statement now - we should be able to do something better using the resource manager.  Maybe something like:

-  For each resource in the resource manager your include a slice of CLI names (and short names).
-  Store these in a map so that you can do a fast locate of the resource-specific manager for the type specified on the CLI
-  Then all this code disappears.

Reply:
It looks like you got as far as adding them to the Res, but just aren't using them.